### DISCUSSION TITLE
How to return custom field in Query.

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2521] [Q&A] [answered]
Hi,
I am using Prisma2(2.0.0-beta.5) with Nexus GraphQL. 
Would like to custom return field in query response which not exist in data model. how we can do that?

Thank you

**Comment by aakashkag** [comment]
Can someone please help?
I am stuck. Lets say I want return field called token which not available in table or data model. how I can return. Currently I am only allow to return fields which I exposed.?

**Reply by Akxe** [reply]
Although I find your starter to be very confusing It proved helpful as to create my own starting point.

**Comment by ryands17** [comment, marked as answer]
Hey @aakashkag :wave: 
You can create a custom field in Nexus as described [here](https://github.com/prisma/prisma-examples/blob/master/typescript/graphql-auth/src/types/AuthPayload.ts) and then you can send that type on any Query/Mutation that you want as described [here](https://github.com/prisma/prisma-examples/blob/126d0d74dc8368aa46366a9683610fb759204f95/typescript/graphql-auth/src/types/Mutation.ts#L8).

**Comment by ludralph** [comment]
Hi @aakashkag,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
